pkg/repository/storage/mongo.go

```diff
@@ -20,19 +26,37 @@ type MongoSSLConfig struct {
 	SSLCertificateAuthoritiyFile string
 }
 
+const (
+	TypeMongoDB    = "mongo"
+	TypeDocDB      = "docdb"
+	DocDBcaFileURI = "https://s3.amazonaws.com/rds-downloads/rds-combined-ca-bundle.pem"
+)
+
 // GetMongoDatabase returns a valid database connection to the configured MongoDB database
-func GetMongoDatabase(dsn, name string, certConfig *MongoSSLConfig) (db *mongo.Database, err error) {
+func GetMongoDatabase(dsn, name, dbType string, certConfig *MongoSSLConfig) (db *mongo.Database, err error) {
 	var mongoOptions *tls.Config
-	if certConfig != nil {
-		mongoOptions, err = options.BuildTLSConfig(map[string]interface{}{
-			"sslClientCertificateKeyFile":     certConfig.SSLClientCertificateKeyFile,
-			"sslClientCertificateKeyPassword": certConfig.SSLClientCertificateKeyFilePassword,
-			"sslCertificateAuthorityFile":     certConfig.SSLCertificateAuthoritiyFile,
-		})
+
+	switch dbType {
+	case TypeMongoDB, "":
+		if certConfig != nil {
+			mongoOptions, err = options.BuildTLSConfig(map[string]interface{}{
+				"sslClientCertificateKeyFile":     certConfig.SSLClientCertificateKeyFile,
+				"sslClientCertificateKeyPassword": certConfig.SSLClientCertificateKeyFilePassword,
+				"sslCertificateAuthorityFile":     certConfig.SSLCertificateAuthoritiyFile,
+			})
+			if err != nil {
+				return nil, fmt.Errorf("could not build SSL config: %w", err)
+			}
+		}
+	case TypeDocDB:
+		mongoOptions, err = getDocDBTLSConfig()
```

Comment:
is this mandatory option? because some users already use DocDB without tls?

Reply:
Or they should use MongoDb in this case?

Reply:
it would work if they would not set a db type, but I'll add a field to make this clear